pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,185 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+
+	"kubevirt.io/client-go/kubecli"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+)
+
+func GetNetworkToIPAMClaimName(virtClient kubecli.KubevirtClient, namespace string, vmiName string, networks []virtv1.Network) (map[string]string, error) {
+	secondaryNetworks := vmispec.FilterNetworksSpec(networks, func(network virtv1.Network) bool {
+		return vmispec.IsSecondaryMultusNetwork(network)
+	})
+
+	networkToIPAMClaimName := make(map[string]string)
+	for _, network := range secondaryNetworks {
+		if allowPersistentIPs, err := getAllowPersistentIPs(virtClient, namespace, network.Multus.NetworkName); err != nil {
+			return map[string]string{}, fmt.Errorf("failed retrieving allowPersistentIPs flag: %w", err)
+		} else if allowPersistentIPs {
+			networkToIPAMClaimName[network.Name] = fmt.Sprintf("%s.%s", vmiName, network.Name)
+		}
+	}
+
+	return networkToIPAMClaimName, nil
+}
+
+func getAllowPersistentIPs(virtClient kubecli.KubevirtClient, namespace string, fullNetworkName string) (bool, error) {
+	namespace, networkName := getNamespaceAndNetworkName(namespace, fullNetworkName)
+	nad, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.Background(), networkName, v1.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to locate network attachment definition %s/%s", namespace, networkName)
+	}
+
+	if nad.Spec.Config == "" {
+		return false, nil
+	}
+
+	netConf := struct {
+		AllowPersistentIPs bool `json:"allowPersistentIPs,omitempty"`
+	}{}
+	err = json.Unmarshal([]byte(nad.Spec.Config), &netConf)
+	if err != nil {
+		return false, fmt.Errorf("failed to unmarshal NAD spec.config JSON: %v", err)
+	}
+
+	return netConf.AllowPersistentIPs, nil
+}
+
+func CreateIPAMClaims(client kubecli.KubevirtClient, vmi *virtv1.VirtualMachineInstance, ownerRef *v1.OwnerReference) error {
+	nonAbsentNetworks := filterNonAbsentNetworks(vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks)
+	networkToIPAMClaimName, err := GetNetworkToIPAMClaimName(client, vmi.Namespace, vmi.Name, nonAbsentNetworks)
+	if err != nil {
+		return fmt.Errorf("failed composing networkToIPAMClaimName: %w", err)
+	}
```

Comment:
we can add the non absent network to the whole filtering related to the the networks that we want so we this will be 
```golang
filteredNetworks, err := filterSecondaryNetworksWithAllowedPersistentIPs(...)
if err != nil {
    return err
}
networkToIpamClaimMap := composeNetworkToIPAMClaimMap(..., filteredNetworks) 
```